testing/opae-c/test_init_c.cpp

```diff
@@ -166,3 +172,274 @@ TEST(init, log_file) {
+    if (stat(cfg_dir_, &st)) {
+      std::string dir = cfg_dir_;
+      // find the first '/' after $HOME
+      int pos = dir.find('/', home.size());
+      while (pos != std::string::npos) {
+        std::string sub = dir.substr(0, pos);
+        // sub is $HOME/<dir1>, then $HOME/<dir1>/<dir2>, ...
+        // if this directory doesn't exist, create it
+        if (stat(sub.c_str(), &st) && sub != "") {
+          ASSERT_EQ(mkdir(sub.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH),
+                    0)
+              << "Error creating subdirectory (" << sub
+              << "}: " << strerror(errno);
+          // keep track of directories created
+          dirs_.push(sub);
+        }
+        pos = pos < dir.size() ? dir.find('/', pos + 1) : std::string::npos;
+      }
+      // finally, we know the entire path didn't exist, create the last
+      // directory
+      ASSERT_EQ(mkdir(cfg_dir_, S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH), 0)
+          << "Error creating subdirectory (" << cfg_dir_
+          << "}: " << strerror(errno);
+      dirs_.push(cfg_dir_);
+    }
+
+    if (stat(cfg_file_.c_str(), &st) == 0) {
+      unlink(cfg_file_.c_str());
+    }
+
+    std::ofstream cfg_stream(cfg_file_);
+    cfg_stream.write(ase_cfg, strlen(ase_cfg));
+    cfg_stream.close();
+
+    setenv("WITH_ASE", "1", 0);
+  }
+
+  virtual void TearDown() override {
+    unsetenv("WITH_ASE");
+    unlink(cfg_file_.c_str());
+    // remove any directories we created in SetUp
+    while (!dirs_.empty()) {
+      unlink(dirs_.top().c_str());
+      dirs_.pop();
+    }
+    // restore the opae_ase.cfg file at OPAE_ASE_CFG_SRC_PATH
+    rename(tmpfile, (char *)OPAE_ASE_CFG_SRC_PATH);
+  }
+
+  char buffer_[PATH_MAX];
+  std::string cfg_file_;
+  char *cfg_dir_;
+  std::stack<std::string> dirs_;
+  std::string src_cfg_file_;
+  char tmpfile[32];
+};
+
+
+/**
+ * @test       find_ase_cfg_2
+ * @brief      Test: find_ase_cfg at OPAE_ASE_CFG_INST_PATH and OPAE_PLATFORM_ROOT
+ *             release location <br>
+ * @details    After renaming a configuration file located in the OPAE_ASE_CFG_SRC_PATH
+ *             and OPAE_ASE_CFG_INST_PATH<br>
+ *             When I call find_ase_cfg
+ *             Then the call is successful<br>
+ */
+TEST_P(init_ase_cfg_p, find_ase_cfg_2) {
+    char *cfg_file = nullptr;
+    std::string inst_cfg_file_;
+    char tmpfile2[32];
+
+    // find_ase_cfg at OPAE_ASE_CFG_INST_PATH
+    cfg_file = find_ase_cfg();
+    EXPECT_NE(cfg_file, nullptr);
+    if (cfg_file)
+        free(cfg_file);
+
+    // copy it to a temporary buffer that we can use dirname with
+    std::string inst_cfg_path = (OPAE_ASE_CFG_INST_PATH? OPAE_ASE_CFG_INST_PATH : "");
+    std::copy(inst_cfg_path.begin(), inst_cfg_path.end(), &buffer_[0]);
+    char *inst_cfg_dir = dirname(buffer_);
+    std::string cfg_dir = (inst_cfg_dir? inst_cfg_dir : "");
+
+    // rename opae_ase.cfg under installation directory
+    strcpy(tmpfile2, "opae_ase.cfg.XXXXXX");
```

Comment:
opae_ase.cfg.XXXXXX doesn't seem to get cleaned up after the test.

Reply:
Did you see test failures? I will double check.

Reply:
No, I just mean that the files get left behind after the tests. You unlink the file in the first test but not the others.

Reply:
It is root caused now.